cpu/msp430/mtarch.c

```diff
@@ -66,10 +66,10 @@ mtarch_start(struct mtarch_thread *t,
 
   t->sp = &t->stack[MTARCH_STACKSIZE - 1];
 
-  *t->sp = (unsigned short)mt_exit;
+  *t->sp = (unsigned short)(void *)mt_exit;
```

Comment:
That's a good question :) If this is not allowed, then having `sp` in `struct mtarch_thread` as `unsigned short *` is wrong, right? Should be `void **` instead or?